php/elFinderVolumeGoogleDrive.class.php

```diff
@@ -0,0 +1,1659 @@
+		list($url) = explode('?', $url);
+		
+		return $url;
+	}
+	
+	/*********************************************************************/
+	/*                        INIT AND CONFIGURE                         */
+	/*********************************************************************/
+
+	/**
+	 * Prepare FTP connection
+	 * Connect to remote server and check if credentials are correct, if so, store the connection id in $ftp_conn
+	 *
+	 * @return bool
+	 * @author Dmitry (dio) Levashov
+	 * @author Cem (DiscoFever)
+	 **/
+	protected function init() {
+		
+		if (!$this->options['googledriveUid']
+		||  !$this->options['accessToken']) {
+			return $this->setError('Required options undefined.');
+		}		
+		
+		// make net mount key
+		$this->netMountKey = md5(join('-', array('googledrive', $this->options['path'])));
+		
+		if (! $this->oauth) {				
+				
+				$client = new Google_Client();
+								 				
+				$client->setAccessToken($this->session->get('elFinderGoogleDriveAuthTokens'));										
+				$this->oauth = new Google_Service_Drive($client);				
+		}
+		
+		if (! $this->oauth) {
+			return $this->setError('OAuth extension not loaded.');
+		}
+				
+		// normalize root path
+		if($this->options['path'] == '/' || $this->options['path'] == 'root'){
+			$this->options['path'] = '/';				
+		}	
+		$this->root = $this->options['path'] = $this->_normpath($this->options['path']);		
+		
+		$this->options['root'] == '' ?  $this->options['root']= 'GoogleDrive.com' : $this->options['root'];
+		
+		if (empty($this->options['alias'])) {
+			$this->options['alias'] = ($this->options['path'] === '/') || ($this->options['path'] === 'root')? $this->options['root'] : $this->getPathtoName($this->options['path']).'@GDrive';
+			
+		}
+
+		$this->rootName = $this->options['alias'];
+				
+		$this->googledriveUid = $this->options['googledriveUid'];
+		$this->tmbPrefix = 'googledrive'.base_convert($this->googledriveUid, 10, 32);
+		
+		if (!empty($this->options['tmpPath'])) {
+			if ((is_dir($this->options['tmpPath']) || @mkdir($this->options['tmpPath'])) && is_writable($this->options['tmpPath'])) {
+				$this->tmp = $this->options['tmpPath'];
+			}
+		}
+		if (!$this->tmp && is_writable($this->options['tmbPath'])) {
+			$this->tmp = $this->options['tmbPath'];
+		}
+		if (!$this->tmp && ($tmp = elFinder::getStaticVar('commonTempPath'))) {
+			$this->tmp = $tmp;
+		}
+		
+		if (!empty($this->options['metaCachePath'])) {
+			if ((is_dir($this->options['metaCachePath']) || @mkdir($this->options['metaCachePath'])) && is_writable($this->options['metaCachePath'])) {
+				$this->metaCache = $this->options['metaCachePath'];
+			}
+		}
+		if (!$this->metaCache && $this->tmp) {
+			$this->metaCache = $this->tmp;
+		}
+		
+		if (!$this->metaCache) {
+			return $this->setError('Cache dirctory (metaCachePath or tmp) is require.');
+		}
+		
+		if (is_null($this->options['syncChkAsTs'])) {
```

Comment:
GoogleDrive driver is not support $this->options['syncChkAsTs'] = true. You must be set $this->options['syncChkAsTs'] = false.